apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V79__add_bfd_and_rda_rbac_roles.sql

```diff
@@ -0,0 +1,106 @@
+/*
+  This migration adds a set of read, write, and migrate roles and groups for BFD and PACA's public/rda schemas.
+
+  If any of these new roles/groups happen to exist, the correct permissions (as defined in the previous migration) will be
+  applied. It does not remove/delete/modify any existing roles, groups, or permissions, which will be cleaned up out-of-band.
+*/
+
+${logic.psql-only} DO $$
+${logic.psql-only} DECLARE
+${logic.psql-only} 	t record;
+${logic.psql-only} BEGIN
+${logic.psql-only}   -- ensure bfd read, write, and migrate roles exists
+${logic.psql-only}   PERFORM create_role_if_not_exists('bfd_reader_role');
+${logic.psql-only}   PERFORM create_role_if_not_exists('bfd_writer_role');
+${logic.psql-only}   PERFORM create_role_if_not_exists('bfd_migrator_role');
+${logic.psql-only}   PERFORM revoke_db_privs('fhirdb', 'bfd_reader_role');
+${logic.psql-only}   PERFORM revoke_db_privs('fhirdb', 'bfd_writer_role');
+${logic.psql-only}   PERFORM revoke_db_privs('fhirdb', 'bfd_migrator_role');
+${logic.psql-only}
+${logic.psql-only}   -- add bfd read, write, migrate roles
+${logic.psql-only}   PERFORM add_reader_role_to_schema('bfd_reader_role', 'public');
+${logic.psql-only}   PERFORM add_writer_role_to_schema('bfd_writer_role', 'public');
+${logic.psql-only}   PERFORM add_migrator_role_to_schema('bfd_migrator_role', 'public');
+${logic.psql-only} 
+${logic.psql-only}   -- add bfd user groups
+${logic.psql-only}   PERFORM add_db_group_if_not_exists('fhirdb', 'bfd_analyst_group');
+${logic.psql-only}   PERFORM add_db_group_if_not_exists('fhirdb', 'bfd_data_admin_group');
+${logic.psql-only}   PERFORM add_db_group_if_not_exists('fhirdb', 'bfd_schema_admin_group');
+${logic.psql-only}   PERFORM revoke_db_privs('fhirdb', 'bfd_analyst_group');
+${logic.psql-only}   PERFORM revoke_db_privs('fhirdb', 'bfd_data_admin_group');
+${logic.psql-only}   PERFORM revoke_db_privs('fhirdb', 'bfd_schema_admin_group');
+${logic.psql-only}   GRANT bfd_reader_role TO bfd_analyst_group;
+${logic.psql-only}   GRANT bfd_writer_role TO bfd_data_admin_group;
+${logic.psql-only}   GRANT bfd_migrator_role TO bfd_schema_admin_group;
```

Comment:
I have some questions about these groups, and the corresponding PACA ones:
1. Who will land in `bfd_data_admin_group` but not in `bfd_schema_admin_group`?
2. Who would get access to just one of PACA or BFD, rather than both?

Reply:
Future proofing along with principle of least-priv. Schema admins can run ddl, data admins cannot. Similar with the groups. Say we had a mitre type scenario, where an external group needs to insert synthetic data. They should not have the ability to view alter other schemas data or schemas (ie paca).

Small inconvenience really.

Reply:
Good enough for me. (Feels a bit analogous to writing a whole bunch of utility code that nothing actually uses yet, though.)

Regardless, my concerns here are around the roles being a bit too complex, not them being insufficient or in any way insecure -- this is not a blocking concern.